hedera-node/src/main/java/com/hedera/services/store/models/Topic.java

```diff
@@ -86,6 +90,66 @@ public static Topic fromGrpcTopicCreate(
 		return topic;
 	}
 
+	/**
+	 * <p>Updates the topic model</p>
+	 * <p><b>Affected properties can be:</b></p>
+	 * <li>Admin key</li>
+	 * <li>Submit key</li>
+	 * <li>Memo</li>
+	 * <li>expirationTimestamp</li>
+	 * <li>autoRenewDurationSeconds</li>
+	 * <li>autoRenewAccountId</li>
+	 * @param op
+	 * 		<p>the Transaction body</p>
+	 * @param designatesAccountRemoval
+	<p>if the auto-renew account is designated for removal, sets the autoRenewAccountId to null.</p>
+	<p>Otherwise - replaces it with the autoRenewAccountId, contained in the transaction body.</p>
+	 */
+	public void update(final ConsensusUpdateTopicTransactionBody op, final boolean designatesAccountRemoval) {
+		updateKeys(op);
+		updateMemo(op);
+		updateExpiry(op);
+		updateAutoRenewPeriod(op);
+		updateAutoRenewAccount(op, designatesAccountRemoval);
+	}
```

Comment:
I don't think the `gRPC` txn body contains too much data. You can check the `TopicCreate` PR for reference. I think the best way would be to explicitly place all the necessary arguments.

Reply:
Given the fact, that the Token.update() method performs several validations like op.hasParticularProperty() - i prefer to use the op argument as an vessel. 
Otherwise - we will pass to the update() method too many arguments, which will decrease the readability of the code.

Reply:
Closing this comment due to duplication with @Daniel-K-Ivanov's comment on the same subject.